support-lib/cpp/Future.hpp

```diff
@@ -379,7 +379,8 @@ class Future {
             constexpr bool await_ready() const noexcept {
                 return false;
             }
-            bool await_suspend(detail::CoroutineHandle<ConcretePromise> finished) const noexcept {
+            template <typename P>
+            bool await_suspend(detail::CoroutineHandle<P> finished) const noexcept {
```

Comment:
@LiFengSC This change lets me derive from Future::promise_type without too much extra work. LMKWYT

Reply:
yeah @jb-gcx is absolutely right.  PromiseTypeBase no longer needs to be templated, when you template this method.

Reply:
good call

Reply:
From my side this is done and can be resolved.